apps/programmes/views.py

```diff
@@ -20,3 +22,22 @@ class ProgrammeViewSet(GenericViewSet, ListModelMixin, RetrieveModelMixin):
 
     filter_backends = [DjangoFilterBackend]
     filterset_class = ProgrammeFilter
+
+class ProgramCoursesView(generics.ListAPIView):
+    serializer_class = CourseSerializer
+
+    def get_queryset(self):
+        programme_uuid = self.kwargs.get('uuid')
+        programme = get_object_or_404(Programme, uuid=programme_uuid)
+        return programme.courses.all()
+    
+    def list(self, request, *args, **kwargs):
+        queryset = self.get_queryset()
+        programme = get_object_or_404(Programme, uuid=self.kwargs.get('uuid'))
+        
+        response_data = {
+            'programme': ProgrammeSerializer(programme).data,
+            'courses': CourseSerializer(queryset, many=True).data
+        }
+        
+        return Response(response_data, status=status.HTTP_200_OK)
```

Comment:
I think its simpler to add a method in the ProgrammeViewSet... For instance

```suggestion
    @action(detail=True, methods=["get"])
    def courses(self, request, *args, **kwargs):
        programme = self.get_object()
        courses = programme.courses.all()
        serializer = CourseSerializer(courses, many=True)
        return Response(serializer.data)
```